### DISCUSSION TITLE
Modal orientations for specific crystal axes only

### DISCUSSION THREAD

**Discussion by ajcross89** [mtex-toolbox/mtex#1709] [Ask Anything]
Hi all,

I'd like to calculate modal orientations ("calcModes") using only a specific crystal axis -- the [0001] axis in quartz -- rather than using the entire ODF. My initial thought was to convert the [0001] axes into vector3d format, calculate their density function, and then run calcModes, like this:

```
% Convert c axis orientations into vector3d format
vec0001 = o_1pg * Miller(0,0,0,1,ebsd('Quartz-new').CS);

% Calculate density function of c axes only
df_1pg_0001 = calcDensity(vec0001,'halfwidth',7.5*degree);

% Find modal c-axis orientations
modes_0001 = calcModes(df_1pg_0001,'numLocal',5);
```

However, calcDensity returns a spherical harmonic function (type: S2FunHarmonic), which is not supported by calcModes, giving the following error:

`Undefined function 'calcModes' for input arguments of type 'S2FunHarmonic'.` 

Does anyone have any ideas on how to acheive this?

Many thanks,

Andy

**Comment by zmichels** [comment]
Are you looking for more than one mode? If one mode is sufficient per set, you can use max() on the output from calcDensity() to get a vector at the maximum density.

**Reply by ajcross89** [reply]
Yeah, I'm looking for the 4-5 most common c-axis orientations

**Comment by kilir** [comment]
Hi,
try `max()` with `'numLocal'` to specify the number of local maxima.
Cheers,
Rüdiger
